pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -157,13 +157,9 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 		podSpec.DNSPolicy = v1.DNSClusterFirstWithHostNet
 	}
 	if osdProps.onPVC() {
-		// The "all" placement is applied separately so it will have lower priority.
-		// We want placement from the storageClassDeviceSet to be applied and override
-		// the "all" placement if there are any overlapping placement settings.
-		c.spec.Placement.All().ApplyToPodSpec(&podSpec)
-		// Apply storageClassDeviceSet PreparePlacement
-		// If nodeAffinity is specified both in the device set and "all" placement,
-		// they will be merged.
+		if !c.spec.Storage.SkipApplyAllPlacement {
+			c.spec.Placement.All().ApplyToPodSpec(&podSpec)
+		}
 		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
 	} else {
 		p := cephv1.GetOSDPlacement(c.spec.Placement)
```

Comment:
For the non-pvc case we would also need to modify `GetOSDPlacement()` to skip the All placement if needed.